src/util/parser.util.ts

```diff
@@ -1,12 +1,27 @@
 import * as Cite from "citation-js";
+import { Reference } from "../model/reference.model";
 
 /**
  * Parse the bibtex data and return an array of reference objects
  * to be later processed by the Data Store.
  * @param data raw bibtex data as a string
  * @returns Array of reference objects
  */
-export function parse (data: string) {
+export function parse (data: string): Reference[] {
     let citation = new Cite(data);
-    return citation.data;
+    let refs: Reference[] = citation.data;
+
+    // Add the more formatted date property instead of relying on "issued"
+    refs = refs.map(ref => {
+        return {
+            ...ref,
+            date: {
+                year: ref.issued["date-parts"][0][0],
+                month: ref.issued["date-parts"][0][1],
+                day: ref.issued["date-parts"][0][2]
+            }
+        }
```

Comment:
I wouldn't bother with adding aliases especially since it means you need to process all the data after parsing, even if it turns out the date is not needed.

Instead you could have a utility function that takes a reference and returns the date in a more usable format. I'd just return an actual Date object, which can then be easily formatted by the calling code. What do you think?

Reply:
- Regarding immutability, I didn't know about that before and indeed it is gonna be interesting to learn about. However, I agree with you that I'm not even over-engineering enough so I should just make it mutable.
- For aliases and date, I agree I should abstract the logic for aliasing and move it into another function that returns a native `Date` Object. Doing so will improve readability and maintenance and prevent the code in the `Parser` to get populated with such details.